packages/flutter_tools/lib/src/build_system/depfile.dart

```diff
@@ -75,7 +75,9 @@ class Depfile {
     for (final File outputFile in files) {
       if (globals.platform.isWindows) {
         // Paths in a depfile have to be escaped on windows.
-        final String escapedPath = outputFile.path.replaceAll(r'\', r'\\');
+        final String escapedPath = outputFile.path
+          .replaceAll(r'\', r'\\')
+          .replaceAll(r' ', r'\ ');
         buffer.write(' $escapedPath');
       } else {
         buffer.write(' ${outputFile.path}');
```

Comment:
Does `_separatorExpr` imply that spaces have to be escaped on all platforms? That is, a space preceded by anything but a '\' is a separator not just on Windows, but on other platforms as well.